Library/Homebrew/extend/ENV/super.rb

```diff
@@ -119,8 +119,11 @@ def determine_path
       end
       paths << apple_gcc42.opt_bin.to_s if apple_gcc42
     when GNU_GCC_REGEXP
-      gcc_formula = gcc_version_formula($&)
-      paths << gcc_formula.opt_bin.to_s
+      begin
+        gcc_formula = gcc_version_formula($&)
+      rescue FormulaUnavailableError
```

Comment:
You could probably wrap the `begin/rescue` around the entire `case` statement as this whole thing is a no-op if the formula isn't found and it's not looped.